plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -289,19 +290,25 @@ class CodeWhispererCodeScanManager(val project: Project) {
         } finally {
             // After code scan
             afterCodeScan(scope)
-            launch {
-                val duration = (Instant.now().toEpochMilli() - startTime).toDouble()
-                CodeWhispererTelemetryService.getInstance().sendSecurityScanEvent(
-                    CodeScanTelemetryEvent(
-                        codeScanResponseContext,
-                        duration,
-                        codeScanStatus,
-                        codeScanResponseContext.payloadContext.srcPayloadSize.toDouble() ?: 0.0,
-                        connection,
-                        scope
+            if (!(
+                    scope == CodeWhispererConstants.CodeAnalysisScope.FILE &&
+                        (!language.isAutoFileScanSupported() || language == CodeWhispererPlainText.INSTANCE)
```

Comment:
this part is a little bit hard to understand
can we try the following instead
1. move this L234 to L225 instead of assigning
https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt#L234

https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt#L225

2. if language is not supported by code scan, return early